### PR TITLE
moving sync_calls to backend

### PR DESCRIPTION
**Pull request by NileshPant1999**

### CODE REVIEW

File: apps/fyle/urls.py [hunk 1]
```diff
@@ -15,4 +16,6 @@
     path('expense_custom_fields/', ExpenseCustomFieldsView.as_view()),
     path('expense_fields/', ExpenseFieldsView.as_view()),
     path('expense_group_settings/', ExpenseGroupSettingsView.as_view()),
+    path('sync_dimensions/', SyncFyleDimensionView.as_view()),
+    path('refresh/', RefreshFyleDimensionView.as_view())
```
Comment (ashwin1111) [inline apps/fyle/urls.py:20]: same

File: apps/fyle/utils.py [hunk 2]
```diff
@@ -301,6 +301,33 @@ def sync_reimbursements(self):
 
         return reimbursement_attributes
 
+    def sync_dimensions(self):
+
+        try:
+            self.sync_categories()
```
Comment (ashwin1111) [inline apps/fyle/utils.py:307]: Follow same order of sync as written in app

File: apps/fyle/utils.py [hunk 3]
```diff
@@ -301,6 +301,33 @@ def sync_reimbursements(self):
...
+        except Exception:
+            pass
```
Comment (ashwin1111) [inline apps/fyle/utils.py:309]: let's log exceptions. We'll not have a way to see if it failed.

File: apps/xero/utils.py [hunk 4]
```diff
@@ -150,7 +149,7 @@ def sync_contacts(self):
 
         for contact in contacts:
             detail = {
-                'email': contact['EmailAddress'] if('EmailAddress' in contact) else None
+                'email': contact['EmailAddress'] if ('EmailAddress' in contact) else None
```
Comment (ashwin1111) [inline apps/xero/utils.py:152]: I think brackets are also not needed

File: apps/xero/utils.py [hunk 5]
```diff
@@ -227,9 +225,35 @@ def create_contact_destination_attribute(self, contact):
             }
         }], self.workspace_id)[0]
 
-        
         return created_contact
 
+    def sync_dimensions(self, workspace_id: str):
+
+        try:
+            self.sync_contacts()
```
Comment (ashwin1111) [inline apps/xero/utils.py:233]: Follow same order of sync as app

**Review by Sravanksk — APPROVED** [review]

File: apps/xero/utils.py [hunk 6]
```diff
@@ -227,9 +227,36 @@ def create_contact_destination_attribute(self, contact):
...
+        except Exception as exception:
+            logger.exception(exception)
+
+        try:
+            self.sync_tracking_categories()
+        except Exception as exception:
+            logger.exception(exception)
+
+        try:
+            self.sync_tenants()
```
Comment (ashwin1111) [inline apps/xero/utils.py:255]: @NileshPant1999 why are we syncing tenants?

**Pull request merged**